### PR TITLE
Redesign landing page

### LINKED ISSUE #1037: landing page: update design for contributors section
Update according to mockups (ref #988)

- Group all the contributor-types, and display the groups horizontally instead of vertically to avoid too much whitespace, and to make it easier to go through the related content without scrolling and losing context.
- Have only one "show affiliations"-button if they're all showing the same content (it's not necessary to have more than one button for the same thing).
- Move "show-affiliations"-button to the very right, bottom of the section to indicate that the extra content is relevant for all the groups, and that it will show up below.

### LINKED ISSUE #1062: landing page: Improve page layout
- Improve the layout by creating a more consistent grid structure
- Make sure each section has a headline and the spacing between sections is consistent
- Make sure the layout is responsive

### PR DESCRIPTION
**Pull request by jennur**
Closes #1062, #1037, #1061, #1046 and #1096 (Existing PR's for these tasks are merged into this branch).

### Changes include
* Responsive layout
* All sections have headlines and consistent spacing.
* Redesign of the details section
    - Details are organized to be in the main section if they can be grouped (.i.e. details that can share a headline goes in the main section under "Additional details".) Single details go in the sidebar under "Details".
    - Upload information is moved to the side bar for it to be more visible.
* Dropdown for export instead of blue links
* Redesign of contributors section. Awaiting inveniosoftware/invenio-rdm-records/issues/802 to merge the affiliation lists and have only one affiliations button.
* Accessibility optimizations (screen reader friendly)

### Desktop

Screenshot 2021-09-22 at 13-59-50 Hanson, White and Hamilton's gallery

Screenshot 2021-09-22 at 13-59-12 Test with lots of content and details Long headline

Screenshot 2021-09-22 at 13-58-39 Testing creators contributors section

### Tablet
Screenshot 2021-09-22 at 14-29-22 Test with lots of content and details Long headline
Screenshot 2021-09-22 at 14-30-16 Hanson, White and Hamilton's gallery

### Mobile
Screenshot 2021-09-22 at 14-33-30 Hanson, White and Hamilton's gallery

### CODE REVIEW

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html [hunk 1]
```diff
@@ -5,123 +5,124 @@
...
-          {% endif %}
-        </div>
-      </div>
-    </div>
+    <dd>
+      {% if identifier.resource_type is defined %}
+      {{ identifier.resource_type.title_l10n }}:
+      {% endif %}
+
+      {{ identifier.identifier + ' ( ' + identifier.scheme | get_scheme_label + ' )' }}
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html:108]: Both "Identifiers" and "Related works" sections should make the identifiers linkable. For that we need ``idutils`` library. Let's talk and I'll show you have to them into links

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html [hunk 2]
```diff
@@ -5,123 +5,124 @@
...
+    <dt class="hidden">References</dt>
   {% for reference in references %}
-    <p>{{ reference.reference + ' ( ' + reference.scheme + ' - ' +  reference.identifier + ' )'}}</p>
+    <dd>{{ reference.reference + ' ( ' + reference.scheme + ' - ' +  reference.identifier + ' )'}}</dd>
   {% endfor %}
 {% endmacro %}
 
+
 {% macro _identifiers_for_group(related_identifiers) %}
+  <dt class="hidden">Related identifiers</dt>
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html:101]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html [hunk 3]
```diff
@@ -5,123 +5,124 @@
...
 {% macro show_funding(funding) %}
   {% for fund in funding %}
     {{ show_detail('Award', fund.award.title + ' ( ' + fund.award.number + ' )' ) if fund.award }}
     {{ show_detail('Funder', fund.funder.name) if fund.funder }}
   {% endfor %}
 {% endmacro %}
 
+
 {% macro show_references(references) %}
+    <dt class="hidden">References</dt>
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html:93]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html [hunk 4]
```diff
@@ -5,123 +5,124 @@
...
-<div class="ui relaxed grid">
-  <div class="two column row">
-    <div class="four wide column">
-      <p><b>{{ title }}</b></p>
-    </div>
-    <div class="twelve wide column">
+  <dt class="ui tiny header">{{ title }}</dt>
+  <dd>
+    {% if title == 'URL' %}
+      <a href="{{value}}" target="_blank" title="Opens in new tab">{{value}}</a>
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html:13]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html [hunk 5]
```diff
@@ -9,54 +9,76 @@
...
-      {{ affiliation[0] }}.
-      {%- if affiliation[2] %}
-        <a href="https://ror.org/{{affiliation[2]}}"><img class="inline-id-icon" src="{{ url_for('static', filename='images/ror-icon.svg') }}"/></a>
-      {%- endif %}
-      {{affiliation[1]}}
-    <br />
+      <dd class="text-muted">
+          {{ affiliation[0] }}.
+          {%- if affiliation[2] %}
+          <a class="identifier-link" href="https://ror.org/{{affiliation[2]}}" aria-label="{{ affiliation[1] }}'s ROR Profile" title="{{ affiliation[1] }}'s ROR Profile">
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html:75]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html [hunk 6]
```diff
@@ -9,54 +9,76 @@
...
+
+{% macro affiliations_accordion(group, affiliations) %}
+<div class="ui sixteen wide tablet three wide computer column title">
+  <div class="dropdown button-wrapper">
+      <button class="ui affiliations-button button mini up">{{ _("show affiliations") }}</button>
+      <button class="ui affiliations-button button mini down">{{ _("hide affiliations") }}</button>
+  </div>
+</div>
+
+<section class="ui sixteen wide column content" aria-label="Affiliations for {{ group }} ">
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html:69]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html [hunk 7]
```diff
@@ -9,54 +9,76 @@
...
+
+  {% for scheme, identifier in creatibutor.person_or_org.identifiers|groupby("scheme") %}
+    {%- if scheme == "orcid" %}
+      {% set identifier_found.value = True %}
+      <a class="identifier-link" href="{{ identifier[0]['identifier']|pid_url('orcid') }}" aria-label="{{ creatibutor.person_or_org.name }}'s ORCID Profile" title="{{ creatibutor.person_or_org.name }}'s ORCID Profile">
+        <img class="inline-id-icon" src="{{ url_for('static', filename='images/orcid.svg') }}" alt="ORCID icon"/>
+      </a>
+    {%- elif scheme == "ror" %}
+      {% set identifier_found.value = True %}
+      <a href="{{ identifier[0]['identifier']|pid_url('ror') }}" aria-label="{{ creatibutor.person_or_org.name }}'s ROR Profile" title="{{ creatibutor.person_or_org.name }}'s ROR Profile">
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html:22]: mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html [hunk 8]
```diff
@@ -9,54 +9,76 @@
...
+        <img class="inline-id-icon" src="{{ url_for('static', filename='images/ror-icon.svg') }}" alt="ROR icon"/>
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html:23]: mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/subjects.html [hunk 9]
```diff
@@ -9,30 +9,23 @@
...
+      <dd class="subject">{{ subject.subject}}</dd>
+    {%- endfor %}
+  </dl>
+  {% endif %}
+
+  {%- for scheme, subjects in subjects|selectattr("scheme")|groupby('scheme')|sort(attribute='scheme') %}
+  <dl class="details-list" aria-label="subjects">
+    <dt class="ui tiny header">{{scheme}}</dt>
+    {%- for subject in subjects %}
+      <dd class="subject">{{ subject.subject}}</dd>
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/subjects.html:27]: Subjects should be linked to a search result page over the subjects. Something along the lines of.

``q=metadata.subjects.subject:<subject>``

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/versions.html [hunk 10]
```diff
@@ -8,11 +8,9 @@
   it under the terms of the MIT License; see LICENSE file for more details.
 -#}
 
-<dl class="ui segment rdm-sidebar">
-  <dt><b>{{ _('Versions')}}</b>
-    <hr class="thin-line">
-  </dt>
-  <dd class="versions">
+<section id="record-versions" aria-label="Record versions" class="ui segment rdm-sidebar">
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/versions.html:11]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/subjects.html [hunk 11]
```diff
@@ -9,30 +9,23 @@
...
+{% if subjects %}
   {% set keywords = subjects|rejectattr("scheme")|list %}
   {% if keywords %}
-  <div class="ui">
-    <div class="subject-scheme column">
-      <h2 class="ui subject header">{{_('Keywords')}}</h2>
-    </div>
-    <div class="subject-label column">
+  <dl class="details-list" aria-label="keywords">
+    <dt class="hidden">Keywords</dt>
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/subjects.html:16]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/upload_info.html [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  Copyright (C) 2020 CERN.
+  Copyright (C) 2020 Northwestern University.
+  Copyright (C) 2021 TU Wien.
+  Copyright (C) 2022 New York University.
+
+  Invenio RDM Records is free software; you can redistribute it and/or modify
+  it under the terms of the MIT License; see LICENSE file for more details.
+#}
+
+<section id="upload-information" aria-label="Upload information" class="ui segment rdm-sidebar">
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/upload_info.html:11]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/keywords_subjects.html [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+  Copyright (C) 2020 Northwestern University.
+  Copyright (C) 2021 TU Wien.
+  Copyright (C) 2022 New York University.
+
+  Invenio RDM Records is free software; you can redistribute it and/or modify
+  it under the terms of the MIT License; see LICENSE file for more details.
+#}
+
+{% if record.metadata.subjects %}
+<section id="keywords-and-subjects" aria-label="Keywords and subjects" class="ui segment rdm-sidebar">
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/keywords_subjects.html:12]: Mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/upload_info.html [hunk 14]
```diff
@@ -0,0 +1,23 @@
+{#
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/side_bar/upload_info.html:1]: I would not include the upload information in the right side bar (seems a bit too prominent). We can talk IRL where to put it.

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/details.html [hunk 15]
```diff
@@ -8,23 +8,219 @@
...
+
+{% if rights %}
+  <div role="tabpanel"
+       class="ui bottom attached {{ 'active' if active }} tab segment"
+       data-tab="licences"
+       aria-labelledby="licences-tab"
+       id="licences"
+       hidden="{{ not active }}"
+  >
+    <dl class="details-list">
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/details.html:177]: I would add the license to the right column. Let's talk how to do it.

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/doi.html [hunk 16]
```diff
@@ -7,28 +7,30 @@
...
+      <div class="header">{{ _("DOI Badge") }}</div>
+      <div class="content">
+        <h4>
+          <small>{{ _("DOI") }}</small>
+        </h4>
+        <h4>
+          <pre>{{ id_doi }}</pre>
+        </h4>
+        {% from "semantic-ui/invenio_formatter/macros/badges.html" import badges_formats_list %}
+        {{ badges_formats_list(url_for('invenio_formatter_badges.badge', title='DOI', value=id_doi, ext='svg', _external=True, _scheme='https'), id_doi|pid_url(scheme='doi')) }}
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/doi.html:31]: mark for translation

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/description.html [hunk 17]
```diff
@@ -7,9 +7,13 @@
...
 {% set description = metadata.get('description') %}
 {% if description %}
+<h2 id="description-heading">{{ _('Description') }}</h2>
 {# description data is being sanitized by marshmallow in the backend #}
-<h2 class="ui header">{{ _('Description')}}</h2>
-<div>{{ description | safe }}</div>
+{{ description | safe }}
 {% endif %}
+
+{{ show_add_descriptions(record.ui.additional_descriptions) if record.ui.additional_descriptions }}
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/description.html:19]: Screenshot 2022-02-03 at 13 53 54

Perhaps simply make them into sections like the description.

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html [hunk 18]
```diff
@@ -9,54 +9,76 @@
...
+
 {% macro show_creatibutors(creatibutors, show_affiliations=False) %}
   {% for creatibutor in creatibutors %}
+  <dd class="creatibutor-wrap">
     {{ creatibutor_icon(creatibutor) }}
-    <a
-      class="ui tooltip-popup text-muted"
+    <a class="ui tooltip-popup text-muted creatibutor-link"
       {% if show_affiliations and creatibutor.affiliations %}
         data-content="{{ creatibutor.affiliations|join('; ', attribute='1') }}"
```
Comment (lnielsen) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/creatibutors.html:41]: ROR icon should be added here as well I think:
Screenshot 2022-02-03 at 15 07 48

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/index.js [hunk 19]
```diff
@@ -51,11 +51,32 @@ if (recordCitationAppDiv) {
...
+      else {
+        $(event.target).accordion('open');
+      }
+    }
+  });
+
+$(".ui.accordion.affiliations-accordion").accordion({
+  selector: {
+    trigger: '.title .affiliations-button'
+  }
```
Comment (lnielsen) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/index.js:75]: Check if this conflicts with https://github.com/inveniosoftware/invenio-app-rdm/pull/1151

**Review by lnielsen — COMMENTED** [review]
Over all looks great - big improvement over previous page!!!

Most comments are on marking string for translations. There's some few comments which might take longer, which we can simply open new issues for and fix them one by one instead of continuing on one big PR.

**Comment by jennur** [comment]
### Updated screenshots

__Desktop__
landing-page_desktop2

__Tablet__
landing-page_tablet2

__Mobile__
landing-page_mobile2

**Pull request merged**